internal/clients/nightfall/nightfall_test.go

```diff
@@ -118,26 +118,92 @@ func (n *nightfallTestSuite) TestScan() {
 	}
 
 	expectedScanReq := createScanReq(detectors, items)
-	expectedScanResp := [][]nightfallAPI.ScanResponse{
-		{},
+	mockAPIClient.EXPECT().ScanPayload(gomock.Any(), gomock.AssignableToTypeOf(expectedScanReq)).Return(expectedScanResponse, nil, nil)
```

Comment:
why are we only checking the type of the 2nd arg passed to `ScanPayload`? Can't we just pass in the actual `expectedScanReq`?

Reply:
Yea it's because in the `Scan` method, there's a `createScanRequest` call that updates the request with the addition of the detectors, so if we pass in the actual `expectedScanReq`, we get this test failure:

```
Got: {[{CREDIT_CARD_NUMBER} {PHONE_NUMBER}] {[this is a string this has a credit card number 4916-6734-7572-5015 tom cruise!!!!!!]}}
Want: is equal to {[{} {}] {[this is a string this has a credit card number 4916-6734-7572-5015 tom cruise!!!!!!]}}
```

Do you have any suggestions on how to resolve this?

Reply:
I believe to get this to work we just need a function like the func `createScanRequest` in `nightfall.go` which takes in the detectors and items to scan and returns a `nightfallAPI.ScanRequest`

Reply:
Isn't that what we're doing on line 120?

Reply:
It's weird how the mock client does expect there to be two detectors, but the values are empty.

Reply:
oh I see, hmmm yea strange

Reply:
Is there a specific reason why detectors is of type `[]*nightfallAPI.Detector`? If I remove the pointer, the tests are now able to read the detector values